domains/transfer-agreements/TransferAgreements.API/API/Controllers/TransferAgreementsController.cs

```diff
@@ -12,24 +13,27 @@ namespace API.Controllers;
 [ApiController]
 public class TransferAgreementsController : ControllerBase
 {
-    private readonly ITransferAgreementService transferAgreementService;
+    private readonly ITransferAgreementRepository transferAgreementRepository;
 
-    public TransferAgreementsController(ITransferAgreementService transferAgreementService) => this.transferAgreementService = transferAgreementService;
+    public TransferAgreementsController(ITransferAgreementRepository transferAgreementRepository) => this.transferAgreementRepository = transferAgreementRepository;
 
     [HttpPost("api/transfer-agreements")]
     public async Task<ActionResult> Create([FromBody] CreateTransferAgreement request)
     {
+        var actor = User.FindActorClaim();
+        var subject = User.FindSubjectClaim();
```

Comment:
As far as I understand, it should only be possible to create transfer agreements for a company and not for a private person. Will this validation be added later?

Reply:
I cannot say, and I would be hesitant to take those kinds of decisions, in a pull request. 

If your question is whether the 2 entities represent persons or companies, then:

 **Subject** represents the _company_
 
 **Actor** represents the _person/user,_ acting in that _companys_ name.